### PR TITLE
Add Couchbase DB Plugin

### PR DESCRIPTION
**Pull request by vbhvmahajan**
This is Couchbase DB plugin for Guardium Universal Connector.

Signed-off-by: Vaibhav Mahajan1 <vaibhav.mahajan1@ibm.com>

### CODE REVIEW

**Review by ofer-haim — APPROVED** [review]

File: filter-plugin/logstash-filter-couchbasedb-guardium/build.gradle [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+// plugin info
+// ===========================================================================
+group                      'com.ibm.guardium.couchbasedb' // must match the package of the main plugin class
+version                    "${file("VERSION").text.trim()}" // read from required VERSION file
+description                = "CouchbaseDB-Guardium filter plugin"
+pluginInfo.licenses        = ['Apache-2.0'] // list of SPDX license IDs
+pluginInfo.longDescription = "This gem is a Logstash Couchbasedb filter plugin required to be installed as part of IBM Security Guardium, Guardium Universal connector configuration. This gem is not a stand-alone program."
+pluginInfo.authors         = ['IBM', 'Vaibhav Mahajan']
+pluginInfo.email           = ['vaibhav.mahajan1@ibm.com']
+pluginInfo.homepage        = "http://www.elastic.co/guide/en/logstash/current/index.html"
```
Comment (Tal-Daniel) [inline filter-plugin/logstash-filter-couchbasedb-guardium/build.gradle:17]: please remove your name, and private email. leave 'IBM' author, and '' in email array.
Consider changing homepage to point to this repository.

File: filter-plugin/logstash-filter-couchbasedb-guardium/filter-test-beats.conf [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	if [type] == "filebeat" and "couchbase" in [tags]{
+		mutate {
+			add_field => { "serverIP" => "%{[@metadata][ip_address]}" }
+			add_field => { "serverHostname" => "%{[agent][hostname]}" }
+			}
+       
+        couchbasedb_guardium_plugin_filter{}
+    }
+    
+   mutate { remove_field => ["serverHostname","@version","@timestamp","type","sequence","message","host","tags","input","log","ecs","agent","serverIP"]}
```
Comment (Tal-Daniel) [inline filter-plugin/logstash-filter-couchbasedb-guardium/filter-test-beats.conf:18]: Please mutate inside IF conditional, as removing fields outside may remove necessary fields from events that should be parsed by other filter plugins. See other plug-ins.

File: filter-plugin/logstash-filter-couchbasedb-guardium/filter-test-generator.conf [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	   	
+filter  {
+       if [type] == 'Couchbase' {
+       mutate {
+			add_field => { "serverIP" => "%{[@metadata][ip_address]}" }
+			add_field => { "serverHostname" => "%{[agent][hostname]}" }
+			}
+        couchbasedb_guardium_plugin_filter{}
+    }
+	mutate { remove_field => ["serverHostname","@version","@timestamp","type","sequence","message","host","tags","input","log","ecs","agent","serverIP"]}
```
Comment (Tal-Daniel) [inline filter-plugin/logstash-filter-couchbasedb-guardium/filter-test-generator.conf:17]: same here

File: filter-plugin/logstash-filter-couchbasedb-guardium/CouchbasedbOverFilebeatPackage/datasourceManifest.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+	"datasourceTypeName":"Couchbase over Filebeat",
+	"inputPluginName":"Filebeat_input",
+	"filterPluginName":"Guardium_CouchbaseDB_filter"
+}
```
Comment (Tal-Daniel) [inline filter-plugin/logstash-filter-couchbasedb-guardium/CouchbasedbOverFilebeatPackage/datasourceManifest.json:5]: Please add "supported_platforms", in another PR (see sibling MongoDB plug-in, for example).

File: filter-plugin/logstash-filter-couchbasedb-guardium/CouchbasedbOverFilebeatPackage/CouchbaseDB/manifest.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{
+	"name": "Guardium_CouchbaseDB_filter",
+	"alias": "Couchbase",
+	"type": "filter",
+	"pipeline_type":"push",
+	"plugin_version": "1.0.0",
+	"datasourceTypes": [{"type":"CouchbaseDB","supportedVersions": ["6.6.2-9600"]}],
+	"supported_input_plugins": ["Filebeat_input"],
+	"developer": "Vaibhav Mahajan",
```
Comment (Tal-Daniel) [inline filter-plugin/logstash-filter-couchbasedb-guardium/CouchbasedbOverFilebeatPackage/CouchbaseDB/manifest.json:9]: Plug-ins developed by IBM employees/sub-contractors should state "IBM" as the developer.

File: filter-plugin/logstash-filter-couchbasedb-guardium/CouchbasedbOverFilebeatPackage/CouchbaseDB/filter.conf [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  if [type] == "filebeat" and "guc_filter_param_datasource_tag" in [tags] {
+    mutate {
+      add_field => { "serverIP" => "%{[@metadata][ip_address]}" }
+      add_field => { "serverHostname" => "%{[agent][hostname]}" }
+    }
+
+    couchbasedb_guardium_plugin_filter{}
+  }
+    
+   mutate { remove_field => ["serverHostname","@version","@timestamp","type","sequence","message","host","tags","input","log","ecs","agent","serverIP"]}
```
Comment (Tal-Daniel) [inline filter-plugin/logstash-filter-couchbasedb-guardium/CouchbasedbOverFilebeatPackage/CouchbaseDB/filter.conf:11]: move mutate inside IF (see comment below).

**Review by Tal-Daniel — CHANGES_REQUESTED** [review]

**Comment by Tal-Daniel** [comment]
Also, Vaibhav, please update available_plugins.md, if possible.

File: docs/available_plugins.md [hunk 7]
```diff
@@ -11,4 +11,5 @@ Pre-packaged plugins can be downloaded from here: https://github.com/IBM/univers
...
 | [PostgreSQL](../filter-plugin/logstash-filter-postgres-guardium/PostgresOverCloudWatchPackage/README.md) | 12.x | AWS | IBM | CloudWatch (pull)   | **Guardium Data Protection only**.<br />Installs required [json_encode](https://www.elastic.co/guide/en/logstash-versioned-plugins/current/v3.0.3-plugins-filters-json_encode.html) filter plug-in. |
 | [Amazon DynamoDB](../filter-plugin/logstash-filter-dynamodb-guardium/README.md) | 2019.11.21 | AWS | IBM | CloudWatch (pull)   | **Guardium Data Protection only** |
-| [Oracle Unified Audit](../filter-plugin/logstash-filter-oua-guardium/README.md)    | 18, 19 | On-prem, RDS                 | IBM      | Oracle Unified Audit (pull) | **Guardium Data Protection only**
+| [Oracle Unified Audit](../filter-plugin/logstash-filter-oua-guardium/README.md)    | 18, 19 | On-prem, RDS                 | IBM      | Oracle Unified Audit (pull) | **Guardium Data Protection only** |
+| [Couchbase](../filter-plugin/logstash-filter-couchbasedb-guardium/README.md)    | 6.6.2-9600        | On-premise / Iaas        | IBM    		| Filebeat (push) |
```
Comment (Tal-Daniel) [inline docs/available_plugins.md:15]: Thanks. Consider stating **Guardium Data Protection only**, if not tested on Guardium Insights yet, like other plug-ins stated, above.

File: filter-plugin/logstash-filter-couchbasedb-guardium/README.md [hunk 8]
```diff
@@ -109,14 +109,17 @@ The Guardium universal connector is the Guardium entry point for native audit lo
...
 		if [type] == "filebeat" and "couchbase" in [tags] {
+			
 			mutate {
 			add_field => { "serverIP" => "%{[@metadata][ip_address]}" }
 			add_field => { "serverHostname" => "%{[agent][hostname]}" }
 			}
+			
 			couchbasedb_guardium_plugin_filter{}
+			
+			mutate { remove_field => ["serverHostname","@version","@timestamp","type","sequence","message","host","tags","input","log","ecs","agent","serverIP"]}
```
Comment (Tal-Daniel) [inline filter-plugin/logstash-filter-couchbasedb-guardium/README.md]: I generally suggest to refer to a conf file, specific section, if possible; follow DRY design pattern (don't repeat yourself), but you can leave it, I guess, if it's short.

**Review by Tal-Daniel — APPROVED** [review]

**Pull request merged**